[CMS] Build errors after updating root master in CMSSW

### Check duplicate issues.

- [x] Checked for duplicates

### Description

We are trying to test the latest root master changes in CMSSW and our build failed with error [a]. ROOT master branch changes we are testing are https://github.com/root-project/root/compare/a607e4242da...96a77141652 . Things worked fine till a607e4242d commit.

Is it something on our side to fix (e.g. adding some missing header ) ?

[a] https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78bbc1/50896/build-logs/PhysicsTools/TagAndProbe/log.html
```
fatal error: module map file '/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/include/ROOT.modulemap' not found
Module ROOT_Foundation_C not found.
Module ROOT_Config not found.
Module ROOT_Rtypes not found.
Module ROOT_Foundation_Stage1_NoRTTI not found.
Module Core not found.
Module Rint not found.
Module RIO not found.
Module MathCore not found.
Module Hist not found.
Module Core not found.
Module Hist not found.
Error in <TClass::RegisterStreamerInfo>: Register StreamerInfo for TObjArray on non-empty slot (3).
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 489
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 87664
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 743
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 2012
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 68378
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 5757
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 136
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 6571
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 11032
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Module Geom not found.
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 20549
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 601
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 45929
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Module Thread not found.
Module RIO not found.
Module Net not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libNet_rdict.pcm file does not exist
Module MultiProc not found.
Module Imt not found.
Module MathCore not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libMathCore_rdict.pcm file does not exist
Module Matrix not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libMatrix_rdict.pcm file does not exist
Module Hist not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libHist_rdict.pcm file does not exist
Module Foam not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libFoam_rdict.pcm file does not exist
Module Graf not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libGraf_rdict.pcm file does not exist
Module GenVector not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libGenVector_rdict.pcm file does not exist
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libGenVector_G__GenVector32_rdict.pcm file does not exist
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 102462
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 42287
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 60305
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 250017
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 3393
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 17431
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 1750948
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 11180
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 13220
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21100
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 83867
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 701
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 1148
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 399131
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 1593
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 66928
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 646
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 29372
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 233
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 33231
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 201964
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 320
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 40393
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 1163
Module ROOT_Foundation_Stage1_NoRTTI not found.
Module RIO not found.
Module Smatrix not found.
Module ROOTTreeViewer not found.
Module ROOTGeomViewer not found.
Module ROOTDataFrame not found.
Module RDAVIX not found.
Module TreePlayer not found.
Module Foam not found.
Module ROOTTreeMap not found.
Module ROOTGraphicsPrimitives not found.
Module RooFitXRooFit not found.
Module Core not found.
Module ROOTHist not found.
Module RooStats not found.
Module Genetic not found.
Module ROOTNTupleUtil not found.
Module ROOTWebDisplay not found.
Module RooFitCore not found.
Module MathMore not found.
Module ROOTHistUtil not found.
Module Thread not found.
Module ROOTGpadv7 not found.
Module Net not found.
Module MathCore not found.
Module Imt not found.
Module NetxNG not found.
Module Minuit2 not found.
Module MultiProc not found.
Module Tree not found.
Module ROOTNTupleBrowse not found.
Module GenVector not found.
Module Minuit not found.
Module ROOTVecOps not found.
Module ROOTEve not found.
Module ROOTFitPanelv7 not found.
Module ROOTTPython not found.
Module ROOTBrowserv7 not found.
Module ROOTBrowsable not found.
Module ROOTNTuple not found.
Module XMLIO not found.
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21171
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 112846
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 104662
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 154
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 259547
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 5509
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 350
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 38848
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 445
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 266040
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 5015
Module Physics not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libPhysics_rdict.pcm file does not exist
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 73612
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 612
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 128150
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 1405
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 142322
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 7205
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 122501
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 7511
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 124028
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 3603
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 13245
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 28881
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 136780
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Module Physics not found.
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 308618
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 23362
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 61558
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 5150
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 196594
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 569
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 32930
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 1883
Module Tree not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libTree_rdict.pcm file does not exist
Module Gpad not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libGpad_rdict.pcm file does not exist
Module TMVA not found.
Module RooFitCore not found.
Error in <TCling::LoadPCM>: ROOT PCM /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/lcg/root/6.39.1-d580e3a35c4b9ed4ccbd4df258ac3422/lib/libRooFitCore_rdict.pcm file does not exist
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 539
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 69266
Error in <TBufferFile::CheckByteCount>: object of class TObjArray read too few bytes: 2 instead of 21
In file included from SimDataFormatsGeneratorProducts_xr dictionary payload:71:
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/src/DataFormats/Common/interface/RefVector.h:14:
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/src/DataFormats/Common/interface/Ref.h:115:
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include/boost/functional.hpp:16:
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include/boost/call_traits.hpp:18:
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include/boost/detail/call_traits.hpp:26:
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include/boost/type_traits/is_arithmetic.hpp:12:
  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include/boost/type_traits/is_integral.hpp:67:41: error: no member named 'ulong_long_type' in namespace 'boost'
 template<> struct is_integral< ::boost::ulong_long_type> : public true_type{};
                               ~~~~~~~~~^
  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include/boost/type_traits/is_integral.hpp:68:41: error: no member named 'long_long_type' in namespace 'boost'
 template<> struct is_integral< ::boost::long_long_type> : public true_type{};
                               ~~~~~~~~~^

 *** Break *** segmentation violation



===========================================================
There was a crash.
This is the entire stack trace of all threads:
===========================================================
#0  0x000014d07bf31e1b in waitpid () from /lib64/libc.so.6
#1  0x000014d07be857c7 in do_system () from /lib64/libc.so.6
#2  0x000014d07b98d785 in TUnixSystem::StackTrace() () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#3  0x000014d07b98d134 in TUnixSystem::DispatchSignals(ESignals) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#4  <signal handler called>
#5  0x000014d07396143b in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#6  0x000014d0705b9451 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#7  0x000014d0705b4c05 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#8  0x000014d0705b4d5b in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#9  0x000014d0705b4c05 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#10 0x000014d0705b6a65 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#11 0x000014d0705b6f2c in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#12 0x000014d070570a45 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#13 0x000014d070570d0f in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#14 0x000014d07054f2c3 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#15 0x000014d0705ef4c0 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#16 0x000014d0705f298b in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#17 0x000014d070550b45 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#18 0x000014d07044684f in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#19 0x000014d07045367b in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#20 0x000014d07045c969 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#21 0x000014d07b912c7d in TClass::LoadClassInfo() const () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#22 0x000014d07b912f10 in TClass::GetListOfBases() () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#23 0x000014d07b91302a in TClass::GetBaseClass(TClass const*) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#24 0x000014d07b9130f8 in TClass::InheritsFrom(TClass const*) const () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#25 0x000014d07b9165a3 in TClass::Property() const () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#26 0x000014d07b910886 in TClass::GetListOfDataMembers(bool) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#27 0x000014d07b91b64a in TClass::GetDataMember(char const*) const () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#28 0x000014d07b95388f in ROOT::Detail::TSchemaRuleSet::AddRule(ROOT::TSchemaRule*, ROOT::Detail::TSchemaRuleSet::EConsistencyCheck, TString*) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#29 0x000014d07b93ada9 in ROOT::TGenericClassInfo::CreateRuleSet(std::vector<ROOT::Internal::TSchemaHelper, std::allocator<ROOT::Internal::TSchemaHelper> >&, bool) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#30 0x000014d07b93b136 in ROOT::TGenericClassInfo::GetClass() () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#31 0x000014d07043674d in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#32 0x000014d07048d327 in ?? () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCling.so
#33 0x000014d07b86b125 in TROOT::RegisterModule(char const*, char const**, char const**, char const*, char const*, void (*)(), std::vector<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int>, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> > > const&, char const**, bool) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/external/el8_amd64_gcc13/lib/libCore.so
#34 0x000014d0796eb341 in (anonymous namespace)::TriggerDictionaryInitialization_SimDataFormatsGeneratorProducts_xr_Impl() [clone .lto_priv.0] () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/lib/el8_amd64_gcc13/libSimDataFormatsGeneratorProducts.so
#35 0x000014d07c5c611a in call_init (l=<optimized out>, argc=argc
entry=3, argv=argv
entry=0x7ffd1441c3e8, env=env
entry=0x7ffd1441c408) at dl-init.c:72
#36 0x000014d07c5c621a in call_init (env=0x7ffd1441c408, argv=0x7ffd1441c3e8, argc=3, l=<optimized out>) at dl-init.c:118
#37 _dl_init (main_map=0x1c45190, argc=3, argv=0x7ffd1441c3e8, env=0x7ffd1441c408) at dl-init.c:119
#38 0x000014d07bfa2aec in _dl_catch_exception () from /lib64/libc.so.6
#39 0x000014d07c5cd92e in dl_open_worker (a=0x7ffd1441b4b0) at dl-open.c:788
#40 dl_open_worker (a=0x7ffd1441b4b0) at dl-open.c:751
#41 0x000014d07bfa2a94 in _dl_catch_exception () from /lib64/libc.so.6
#42 0x000014d07c5cdb81 in _dl_open (file=0x1c43fa0 "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/tmp/el8_amd64_gcc13/src/PhysicsTools/TagAndProbe/plugins/PhysicsToolsTagAndProbe_plugins/libPhysicsToolsTagAndProbe_pl"..., mode=-2147483391, caller_dlopen=0x14d07c71d615 <edmplugin::SharedLibrary::SharedLibrary(std::filesystem::__cxx11::path const&)+69>, nsid=<optimized out>, argc=3, argv=<optimized out>, env=0x7ffd1441c408) at dl-open.c:870
#43 0x000014d07b3fcf8a in dlopen_doit () from /lib64/libdl.so.2
#44 0x000014d07bfa2a94 in _dl_catch_exception () from /lib64/libc.so.6
#45 0x000014d07bfa2b53 in _dl_catch_error () from /lib64/libc.so.6
#46 0x000014d07b3fd52e in _dlerror_run () from /lib64/libdl.so.2
#47 0x000014d07b3fd02a in dlopen

GLIBC_2.2.5 () from /lib64/libdl.so.2
#48 0x000014d07c71d615 in edmplugin::SharedLibrary::SharedLibrary(std::filesystem::__cxx11::path const&) () from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_16_1_ROOT6_X_2026-01-25-2300/lib/el8_amd64_gcc13/libFWCorePluginManager.so
#49 0x000000000040ca8f in main::{lambda()#1}::operator()() const ()
#50 0x0000000000405e6c in main ()
===========================================================
```

### Reproducer

One need to build CMS Offline software (CMSSW)

### ROOT version

ROOT master branch

### Installation method

build from sources

### Operating system

AlmaLinux 

### Additional context

Building using GCC 13.4.0

--- a/core/base/src/TROOT.cxx
+++ b/core/base/src/TROOT.cxx
@@ -3159,7 +3159,26 @@ const TString &TROOT::GetSharedLibDir()
 
       fs::path p = info->dlpi_name;
       if (p.filename() == _R_QUOTEVAL_(LIB_CORE_NAME)) {
-         libdir = p.parent_path().c_str();
+         std::error_code ec;
+
+         // Resolve symlinks: critical for environments like CMSSW, where the
+         // ROOT libraries are loaded via symlinks that point to the actual
+         // install directory
+         fs::path resolved = fs::canonical(p, ec);
+         if (ec) {
+            ::Error("TROOT",
+                    "Failed to canonicalize detected ROOT shared library path:\n"
+                    "%s\n"
+                    "Error code: %d (%s)\n"
+                    "Error category: %s\n"
+                    "This is an unexpected internal error and ROOT might not work.\n"
+                    "Please report this issue on GitHub: https://github.com/root-project/root/issues",
+                    p.string().c_str(), ec.value(), ec.message().c_str(), ec.category().name());
+            // Fall back to the loader path if canonicalization fails. The path
+            // will likely be wrong, but at least not garbage
+            resolved = p;
+         }
+         libdir = resolved.parent_path().c_str();
          return 1; // stop iteration
       }
       return 0; // continue

